pkgs/servers/matrix-synapse/matrix-appservice-irc/default.nix

```diff
@@ -7,6 +7,27 @@ let
   };
   version = (lib.importJSON ./package.json).version;
   srcInfo = lib.importJSON ./src.json;
+
+  # TODO: package matrix-rust-sdk and use that instead of fetching & patching binaries
+  platformString =
+    if stdenv.hostPlatform.system == "x86_64-linux" then "linux-x64-gnu"
+    else if stdenv.hostPlatform.system == "aarch64-linux" then "linux-arm64-gnu"
+    else throw "matrix-sdk-crypto-nodejs: Architecture not supported";
```

Comment:
There are more platforms supported upstream, but I'm reluctant to add them since I can't test them and constrcuts like this are a maintenance nightmare anyway.

https://github.com/matrix-org/matrix-rust-sdk/releases/